src/App.tsx

```diff
@@ -15,48 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+  DEFAULT = '',
+}
+
+function getOrderedGoods(
+  goods: string[],
+  orderedMethod: SortType,
+  isReversed: boolean,
+) {
+  const finalGoods = [...goods];
+
+  switch (orderedMethod) {
+    case SortType.ALPHABET:
+      finalGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+
+    case SortType.LENGTH:
+      finalGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    finalGoods.reverse();
+  }
+
+  return finalGoods;
+}
+
 export const App: React.FC = () => {
+  const [orderMethod, setOrderMethod] = useState(SortType.DEFAULT);
+  const [isReversed, setIsReversed] = useState(false);
+  const orderedGoods = getOrderedGoods(
+    goodsFromServer,
+    orderMethod,
+    isReversed,
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info', {
+            'is-light': orderMethod !== SortType.ALPHABET,
+          })}
+          onClick={() => setOrderMethod(SortType.ALPHABET)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button is-success', {
+            'is-light': orderMethod !== SortType.LENGTH,
+          })}
+          onClick={() => setOrderMethod(SortType.LENGTH)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setIsReversed(!isReversed)}
```

Comment:
you should update it with function inside ([article](https://react.dev/reference/react/useState#setstate-caveats)):
```suggestion
          onClick={() => setIsReversed((isReversed) => !isReversed}
```